libraries/AP_BattMonitor/AP_BattMonitor_Analog_GPIO.cpp

```diff
@@ -30,20 +39,32 @@ void AP_BattMonitor_Analog_GPIO::timer() {
     return;
   }
 
-  bool new_is_using_battery = (bool)((buf & 0x04) == 0);
-  if(!_is_using_battery && new_is_using_battery) {
-    gcs().send_text(MAV_SEVERITY_CRITICAL, "Using battery power");
-  } else if (_is_using_battery && !new_is_using_battery) {
-    gcs().send_text(MAV_SEVERITY_CRITICAL, "Using tether power");
+  bool new_is_using_battery = (bool)((buf & _use_tether_mask) == 0); // Bit 1 of register 0 is FET_EN_TETHER.
+  if(_is_using_battery ^ new_is_using_battery) {
+    if (new_is_using_battery) {
+      gcs().send_text(MAV_SEVERITY_CRITICAL, "Using battery power");
+    } else {
+      gcs().send_text(MAV_SEVERITY_CRITICAL, "Using tether power");
+    }
   }
 
   //The state of the battery discharging is on GPIO 0
   _is_using_battery = new_is_using_battery;
   _state.on_tether_power = !_is_using_battery;
 }
 
+uint8_t AP_BattMonitor_Analog_GPIO::get_I2C_addr(const AP_BattMonitor_Params::BattMonitor_Type type)
+{
+  if(type == AP_BattMonitor_Params::BattMonitor_TYPE_ANALOG_VOLTAGE_AND_CURRENT_AND_GPIO_REV2)
+    return AP_BATTMONITOR_ANALOG_GPIO_I2C_ADDR_REV2;
+  else if(type == AP_BattMonitor_Params::BattMonitor_TYPE_ANALOG_VOLTAGE_AND_CURRENT_AND_GPIO_REV3)
+    return AP_BATTMONITOR_ANALOG_GPIO_I2C_ADDR_REV3;
+  else
+    return 0x00;
```

Comment:
Can we add an error log message here?

Reply:
We could add a `gcs.status` string, but this is actually run at boot-up so you won't get the msg by the time it is run. Also this is only called when the param are rev2 or rev3: 
https://github.com/planckaero/ardupilot/blob/5cc33f0fd8eab3b8cc9c1c91ee73cb4d527ba3c9/libraries/AP_BattMonitor/AP_BattMonitor.cpp#L176-L183